src/NuGet.Core/NuGet.Protocol/Providers/PackageUpdateResourceV3Provider.cs

```diff
@@ -30,7 +29,11 @@ public PackageUpdateResourceV3Provider()
             if (serviceIndex != null)
             {
                 var baseUrl = serviceIndex.GetServiceEntryUri(ServiceTypes.PackagePublish);
-
+                // Check for a not HTTPS source
+                if (baseUrl.Scheme == Uri.UriSchemeHttp && baseUrl.Scheme != Uri.UriSchemeHttps && !source.PackageSource.AllowInsecureConnections)
```

Comment:
Redundant condition. There are a few more occurrences of checking for that the scheme is equal to both http and not https in this PR so this comment applies to all those instances.